packages/test/src/test-local-activities.ts

```diff
@@ -0,0 +1,312 @@
+        async succeedAfterFirstAttempt() {
+          attempts++;
+          if (attempts === 1) {
+            throw new Error('Retry me please');
+          }
+        },
+      },
+    });
+
+    await runWorker(worker, async () => {
+      const handle = await client.start(workflows.throwAnErrorWithBackoff, {
+        workflowId: uuid4(),
+        taskQueue,
+        workflowTaskTimeout: '3s',
+      });
+      await handle.result();
+      const { history } = await client.service.getWorkflowExecutionHistory({
+        namespace: 'default',
+        execution: { workflowId: handle.workflowId },
+      });
+      const timers = history?.events?.filter(
+        (ev) => ev.eventType === temporal.api.enums.v1.EventType.EVENT_TYPE_TIMER_FIRED
+      );
+      t.is(timers?.length, 1);
+
+      const markers = history?.events?.filter(
+        (ev) => ev.eventType === temporal.api.enums.v1.EventType.EVENT_TYPE_MARKER_RECORDED
+      );
+      t.is(markers?.length, 2);
+    });
+  });
+
+  test('Local activity can be intercepted', async (t) => {
+    const { client, taskQueue } = t.context;
+    const worker = await Worker.create({
+      taskQueue,
+      workflowsPath: require.resolve('./workflows/local-activity-testers'),
+      // Interceptors included with workflow implementations
+      interceptors: {
+        workflowModules: [require.resolve('./workflows/local-activity-testers')],
+        activityInbound: [
+          () => ({
+            async execute(input, next) {
+              t.is(defaultPayloadConverter.fromPayload(input.headers.secret), 'shhh');
+              return await next(input);
+            },
+          }),
+        ],
+      },
+      activities,
+    });
+    await runWorker(worker, async () => {
+      const res = await client.execute(workflows.runOneLocalActivity, {
+        workflowId: uuid4(),
+        taskQueue,
+        args: ['message'],
+      });
+      t.is(res, 'messagemessage');
+    });
+  });
+
+  // TODO: fix Core shutdown and reenable this test
+  test.skip('Worker shutdown while running a local activity completes after completion', async (t) => {
+    const { client, taskQueue } = t.context;
+    const subj = new Subject<void>();
+    const worker = await Worker.create({
+      taskQueue,
+      activities,
+      workflowsPath: require.resolve('./workflows/local-activity-testers'),
+      interceptors: {
+        workflowModules: [require.resolve('./workflows/local-activity-testers')],
+      },
+      sinks: {
+        test: {
+          timerFired: {
+            fn() {
+              subj.next();
+            },
+          },
+        },
+      },
+      // Just in case
+      shutdownGraceTime: '10s',
+    });
+    const handle = await client.start(workflows.cancelALocalActivity, {
+      workflowId: uuid4(),
+      taskQueue,
+      workflowTaskTimeout: '3s',
+    });
+    const p = worker.run();
+    await firstValueFrom(subj);
+    worker.shutdown();
```

Comment:
Since firstValueFrom resolves when timer is done, and when timer is done, activity is canceled, does the cancel or the shutdown happen first? 

(I'm wanting to make sure that the activity is still running so that this test tests what it's try to test 😄)

Reply:
We shutdown Core while just before requesting to cancel the local activity.
We expect it to shutdown once the workflow and local activity are complete.